syntax = "proto3";

package notificationPackage;

service NotificationService {
  rpc CreateNotification (CreateNotificationRequest) returns (NotificationResponse);
  rpc GetNotification (GetNotificationRequest) returns (NotificationResponse);
  rpc GetNotifications (GetNotificationsRequest) returns (GetNotificationsResponse);
  rpc UpdateNotification (GetNotificationRequest) returns (NotificationResponse);
  rpc DeleteNotification (GetNotificationRequest) returns (DeleteNotificationResponse);
}

message Notification {
  string notification_id = 1;
  string sender_id = 2;
  string receiver_id = 3;
  string message = 4;
  string status = 5; // read/unread
  string created_at = 6;
  string updated_at = 7;
}

message CreateNotificationRequest {
  string sender_id = 1;
  string receiver_id = 2;
  string scenario = 3;
}

message NotificationResponse {
  string notification_id = 1;
  string sender_id = 2;
  string receiver_id = 3;
  string message = 4;
  string status = 5; // read/unread
  string created_at = 6;
  string updated_at = 7;
}

message GetNotificationRequest {
  string notification_id = 1;
}

message GetNotificationsRequest {
  string receiver_id = 1;
  string status = 2;
}

message GetNotificationsResponse {
  repeated NotificationResponse notifications = 1;
}

message DeleteNotificationResponse {
  bool success = 1;
}
